### PR TITLE
feat: expose d3-shape interpolation props

### LINKED ISSUE #399: Feature request: Expose Recharts / d3-shape interpolation
Hey all, 

Thanks for creating this awesome lib! I was looking into implementing some line and area charts and I was amazed by how simple this is using Tremor.

One small request from my side, would it be possible to export Rechart's interpolation types so we can get smooth points in area and line charts? 
```
[type](https://recharts.org/en-US/api/Area#type)'basis' | 'basisClosed' | 'basisOpen' | 'linear' | 'linearClosed' | 'natural' | 'monotoneX' | 'monotoneY' | 'monotone' | 'step' | 'stepBefore' | 'stepAfter' | Function
```

Is this something you have in mind? I would be more than happy to help with this if needed.

Best,

### PR DESCRIPTION
**Pull request by dani-z**
Fixes #399

### CODE REVIEW

File: src/components/chart-elements/common/BaseChartProps.tsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ interface BaseChartProps extends React.HTMLAttributes<HTMLDivElement> {
   autoMinValue?: boolean;
   minValue?: number;
   maxValue?: number;
+  curveType?: CurveType;
```
Comment (mitrotasios) [inline src/components/chart-elements/common/BaseChartProps.tsx]: I wouldn't add the prop to the BaseChartProps as the interface is shared with the `BarChart` as well, which would not support a `curveType`

**Review by mitrotasios — CHANGES_REQUESTED** [review]
Thanks for the work! Just a minor change.

File: src/components/chart-elements/common/BaseChartProps.tsx (hunk 1 above)
Comment (dani-z) [reply src/components/chart-elements/common/BaseChartProps.tsx]: You're right, I just realised that those props are shared. I will amend the PR to only use this props for line and area charts.

Comment (dani-z) [reply src/components/chart-elements/common/BaseChartProps.tsx]: Sorry for the delay @mitrotasios, I have updated the PR as suggested.

Comment (mitrotasios) [reply src/components/chart-elements/common/BaseChartProps.tsx]: no worries, thanks a lot 🤗

**Review by mitrotasios — APPROVED** [review]

**Comment by dani-z** [comment]
Hey @mitrotasios, thanks for merging. I've just seen that in the compiled code, the `CurveType` is wrongly typed, unfortunately :( 
image

However, the type does work, I am wondering if there's some naming conflict with the props?

**Comment by mitrotasios** [comment]
hey @dani-z, that's right. We wanted to restrict the curveType to only contain one type for each category. We deemed the other available types such as `basisClosed` as unnecessary. Is there a specific type you're missing?

**Comment by dani-z** [comment]
Yeah, I was looking for smooth edges, which is `monotone`. What are the benefits of restricting it? If the platform behind supports it, you could easily surface that. I do agree, however, that some of the options don't create visible UI differences

**Comment by mitrotasios** [comment]
I see. The reason for restricting is that tremor currently aims at keeping things as simple as possible. In our opinion having one curve type for each category, i.e. "curved", linear and step, is enough as there is no real visual benefit in having different curved charts (e.g. monotone vs natural).

**Comment by dani-z** [comment]
Sure, keeping things simple is a good decision. I am unsure, though, how exposing something that is already in the underneath layer would complicate it. But again, I appreciate your decision. 🙏

**Pull request merged**